syntax = "proto3";


service ConversationIndex {
    rpc getCitations(getConvCitationsRequest) returns (getConvCitationsResponse) {}
    rpc getIndex(getIndexRequest) returns (getIndexResponse) {}
    rpc deleteIndex(deleteIndexRequest) returns (deleteIndexResponse) {}
}

message getConvCitationsRequest {
    string conversationId = 1;
    string query = 2;
    int32 k = 3;
}

message getConvCitationsResponse {
    repeated ConvCitation citations = 1;
}

message ConvCitation {
    string filename = 1;
    int32 pagenum = 2;
    string text = 3;
}

message getIndexRequest {
    string conversationId = 1;
    string fileName = 2;
}

message getIndexResponse {

    enum IndexStatus {
        INDEXED = 0;
        NOT_FOUND = 1;
    }

    string conversationId = 1;
    string fileName = 2;
    IndexStatus status = 3;
}

message deleteIndexRequest {
    string conversationId = 1;
    string fileName = 2;
}

message deleteIndexResponse {

    enum DeletionStatus {
        DELETED = 0;
        NOT_FOUND = 1;
        FAILED = 2;
    }

    string conversationId = 1;
    string fileName = 2;
    DeletionStatus status = 3;
}
